# Trends and Predictions in Netflix's Daily Top 10 Rankings

## Data Wrangling

In this section, we will prepare our dataset for analysis by addressing inconsistencies and ensuring data quality. The dataset includes Netflix's daily top 10 rankings for movies and TV shows in the United States from 2020 to March 2022, featuring attributes like title, release date, type, and viewership metrics.

Our focus will be on handling missing values, standardizing data types, and rectifying anomalies to ensure the data is clean and consistent. This foundational step is crucial for accurate analysis and model building, enabling us to explore trends, identify key factors influencing viewership scores, and develop predictive models to forecast rankings.

In [1]:
# Import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime

In [3]:
# Load the dataset
df = pd.read_csv('netflix daily top 10.csv')

# Display the first few rows of the dataframe
df.head()


As of  Rank Year to Date Rank Last Week Rank  \
0  2020-04-01     1                 1              1   
1  2020-04-01     2                 2              -   
2  2020-04-01     3                 3              2   
3  2020-04-01     4                 4              -   
4  2020-04-01     5                 5              4   

                          Title     Type Netflix Exclusive  \
0  Tiger King: Murder, Mayhem …  TV Show               Yes   
1                         Ozark  TV Show               Yes   
2                  All American  TV Show               NaN   
3                  Blood Father    Movie               NaN   
4                  The Platform    Movie               Yes   

  Netflix Release Date  Days In Top 10  Viewership Score  
0         Mar 20, 2020               9                90  
1         Jul 21, 2017               5                45  
2         Mar 28, 2019               9                76  
3         Mar 26, 2020               5                30  
4         Mar 20, 2020               9                55

In [4]:
# Check for duplicate rows
duplicates = df.duplicated()
print(f"Number of duplicate rows: {duplicates.sum()}")

# Drop duplicates if any
df = df.drop_duplicates()


Number of duplicate rows: 0


In [10]:
# Check data types and basic statistics
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7100 entries, 0 to 7099
Data columns (total 10 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   As of                 7100 non-null   datetime64[ns]
 1   Rank                  7100 non-null   int64         
 2   Year to Date Rank     7100 non-null   object        
 3   Last Week Rank        7100 non-null   object        
 4   Title                 7100 non-null   object        
 5   Type                  7100 non-null   object        
 6   Netflix Exclusive     4599 non-null   object        
 7   Netflix Release Date  7100 non-null   datetime64[ns]
 8   Days In Top 10        7100 non-null   int64         
 9   Viewership Score      7100 non-null   int64         
dtypes: datetime64[ns](2), int64(3), object(5)
memory usage: 554.8+ KB


In [11]:
df.describe()

As of         Rank  \
count                           7100  7100.000000   
mean   2021-03-21 12:00:00.000000256     5.500000   
min              2020-04-01 00:00:00     1.000000   
25%              2020-09-25 00:00:00     3.000000   
50%              2021-03-21 12:00:00     5.500000   
75%              2021-09-15 00:00:00     8.000000   
max              2022-03-11 00:00:00    10.000000   
std                              NaN     2.872484   

                Netflix Release Date  Days In Top 10  Viewership Score  
count                           7100     7100.000000       7100.000000  
mean   2020-06-21 08:05:32.619718144       24.123662        122.790141  
min              2007-01-15 00:00:00        1.000000          1.000000  
25%              2020-04-26 00:00:00        3.000000         19.000000  
50%              2020-10-02 00:00:00        7.000000         50.000000  
75%              2021-05-14 00:00:00       18.000000        128.000000  
max              2022-03-04 00:00:00      428.000000       1474.000000  
std                              NaN       58.473789        213.861642

In [13]:
# Check for any syntactic rule violations (e.g., date format)
# Convert 'As of' column to datetime
df['As of'] = pd.to_datetime(df['As of'], format='%Y-%m-%d')

# Ensure 'Netflix Release Date' follows the same format
df['Netflix Release Date'] = pd.to_datetime(df['Netflix Release Date'], errors='coerce')

# Check for any semantic rule violations
# Ensure 'Rank' is between 1 and 10
assert df['Rank'].between(1, 10).all(), "Rank column contains values outside 1-10"

In [14]:
# Check for missing values
missing_values = df.isnull().sum()
print("Missing values in each column:\n", missing_values)


Missing values in each column:
 As of                      0
Rank                       0
Year to Date Rank          0
Last Week Rank             0
Title                      0
Type                       0
Netflix Exclusive       2501
Netflix Release Date       0
Days In Top 10             0
Viewership Score           0
dtype: int64


In [17]:
# Check the percentage of missing values in 'Netflix Exclusive'
missing_percentage = df['Netflix Exclusive'].isnull().mean() * 100
print(f"Percentage of missing values in 'Netflix Exclusive': {missing_percentage:.2f}%")


Percentage of missing values in 'Netflix Exclusive': 35.23%


In [18]:
# Fill missing values in 'Netflix Exclusive' with the mode
mode_value = df['Netflix Exclusive'].mode()[0]
df['Netflix Exclusive'].fillna(mode_value, inplace=True)

# Verify there are no more missing values
missing_values_after = df['Netflix Exclusive'].isnull().sum()
print(f"Missing values in 'Netflix Exclusive' after imputation: {missing_values_after}")


Missing values in 'Netflix Exclusive' after imputation: 0


#### Summary

In this data wrangling section, we systematically prepared the Netflix daily top 10 rankings dataset for analysis. We started by loading the dataset and inspecting its initial structure, followed by removing duplicate rows to ensure data integrity. We then checked and corrected data types, particularly for date columns, and verified that the 'Rank' column contained valid values between 1 and 10. To handle missing values in the "Netflix Exclusive" column, which had 35.23% missing data, we imputed the missing entries with the column's mode. These steps ensured that our dataset is clean, consistent, and ready for further analysis.